### PR TITLE
Equal constraint using comparer

### LINKED ISSUE #1897: EqualTo().Using() prevents caller from comparing strings to anything else
Given a suitable `DateTimeComparer` class that converts `IConvertible` objects to `DateTime`, I expect this code to not throw an assertion:
```
Assert.That(new DateTime(2016, 8, 2), Is.EqualTo("2 August 2016")
                                        .Using(DateTimeComparer.Instance));
```
But it does, because `Using(DateTimeCompare.Instance)`  returns either a `ComparerAdapter` or an `EqualityComparerAdapter`, neither of which override the default implementation of `CanCompare`.

Since the person writing the test is taking responsibility for comparing the actual and expected values (due to calling `Using`), `CanCompare` should always return `true` for any `ExternalComparer` added when calling `Using`.

The original issue and MCVE is described at [Stackoverflow](http://stackoverflow.com/q/38711296/3195526).

### PR DESCRIPTION
**Pull request by manfred-brands**
Fixes #1897

Changes EqualityAdapter.CanCompare if one of the arguments is `string` and the other is not `IEnumerable`.

### CODE REVIEW

File: src/NUnitFramework/framework/Constraints/EqualityAdapter.cs
Comment (mikkelbu) [inline src/NUnitFramework/framework/Constraints/EqualityAdapter.cs:205]: What is the reason for introducing this override? (As far as I can tell it is overriding the same implementation in the parent class `GenericEqualityAdapter<T>`)

File: src/NUnitFramework/framework/Constraints/EqualityAdapter.cs
Comment (mikkelbu) [inline src/NUnitFramework/framework/Constraints/EqualityAdapter.cs:241]: Same comment as above

**Review by mikkelbu — APPROVED** [review]
Looks good to me, but we will need a second review before merging this (as per the rules of the repository).

File: src/NUnitFramework/framework/Constraints/EqualityAdapter.cs
Comment (jnm2) [inline src/NUnitFramework/framework/Constraints/EqualityAdapter.cs:62]: If desired, the whole method impl could be:

```cs
return (x is string || !(x is IEnumerable))
    && (y is string || !(y is IEnumerable));
```

**Review by jnm2 — DISMISSED** [review]

File: src/NUnitFramework/framework/Constraints/EqualityAdapter.cs
Comment (manfred-brands) [reply src/NUnitFramework/framework/Constraints/EqualityAdapter.cs]: > What is the reason for introducing this override? (As far as I can tell it is overriding the same implementation in the parent class `GenericEqualityAdapter<T>`)

Initially I was overriding all `CanCompare` operations as per idea in the original ticket. These were indeed useless.
I have added a commit to remove the new overloads again. Thanks for detecting this.

File: src/NUnitFramework/framework/Constraints/EqualityAdapter.cs
Comment (manfred-brands) [reply src/NUnitFramework/framework/Constraints/EqualityAdapter.cs]: Removed.

File: src/NUnitFramework/framework/Constraints/EqualityAdapter.cs
Comment (manfred-brands) [reply src/NUnitFramework/framework/Constraints/EqualityAdapter.cs]: @jnm2 I like it, added a commit to the branch doing so.

**Review by jnm2 — APPROVED** [review]
Thanks for working on this!

**Review by mikkelbu — APPROVED** [review]
Also looks good to me. Thanks for doing this @manfred-brands. I'll wait with the merge in case you want to clean up the commits as in the Analyzer project.

**Comment by manfred-brands** [comment]
Thanks @mikkelbu. Cleaned up commits.

**Review by mikkelbu — APPROVED** [review]

**Pull request merged**
